src/Microsoft.DocAsCode.Dfm/JsonRenderer.cs

```diff
@@ -0,0 +1,290 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Dfm
+{
+    using System.Collections.Immutable;
+    using System.Text;
+
+    using Microsoft.DocAsCode.MarkdownLite;
+
+    public class JsonRenderer : HtmlRenderer
+    {
+        public enum Type
+        {
+            size,
+            child
+        }
+
+        public ImmutableDictionary<string, string> Tokens { get; set; }
+
+        public string Insert(string name, string tokenContent, Type tokentype)
+        {
+            var result = (StringBuffer)"{\n\"name\":\"" + name + "\"";
+            if (tokentype == Type.child)
+            {
+                // If tokenContent is not empty ,should remove the last character(',')
+                if (tokenContent.Length > 0)
+                {
+                    tokenContent = tokenContent.Remove(tokenContent.Length - 1);
+                }
+
+                result += ",\n\"contents\":[\n" + tokenContent + "]\n},";
+            }
+            else if (tokentype == Type.size)
+            {
+                // size will be need later
+                // result += "\"size\":" + tokenContent + "\n},";
+                result += "\n},";
+            }
+            return result;
+        }
+
+        public string GetSize()
+        {
+            return "1";
+        }
+
+        #region Block
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownNewLineBlockToken token, MarkdownBlockContext context)
+        {
+            // do nothing.
+            return StringBuffer.Empty;
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownCodeBlockToken token, MarkdownBlockContext context)
+        {
+            return this.Insert("Code(" + token.Lang + ")", this.GetSize(), Type.size);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownHeadingBlockToken token, MarkdownBlockContext context)
+        {
+            string level = token.Depth.ToString();
+            var childContent = new StringBuilder();
```

Comment:
as far as I know , StringBuilder is faster than StringBuffer (but StringBuffer is synchronized),do we need 'synchronized'?

---
In reply to: [74544144](https://github.com/dotnet/docfx/pull/602#discussion_r74544144) [](ancestors = 74544144)